support/cas-server-support-gua/src/main/java/org/apereo/cas/gua/impl/LdapUserGraphicalAuthenticationRepository.java

```diff
@@ -20,12 +21,24 @@
  * @since 5.1.0
  */
 @Slf4j
-@RequiredArgsConstructor
-public class LdapUserGraphicalAuthenticationRepository implements UserGraphicalAuthenticationRepository {
+public class LdapUserGraphicalAuthenticationRepository implements UserGraphicalAuthenticationRepository, DisposableBean {
     private static final long serialVersionUID = 421732017215881244L;
 
     private final CasConfigurationProperties casProperties;
 
+    private ConnectionFactory connectionFactory;
+
+    public LdapUserGraphicalAuthenticationRepository(final CasConfigurationProperties casProperties) {
+        this.casProperties = casProperties;
+        this.connectionFactory = LdapUtils.newLdaptiveConnectionFactory(
+            casProperties.getAuthn().getGua().getLdap());
+    }
```

Comment:
Pass the connection factory, remove ctor and use lombok.

Reply:
Do you want to convert `connectionFactory` to a separate bean? I think this is a overkill to create a separate bean just to wrap `connectionFactory` into a `DisposableBean`.

Reply:
Not quite. Let me clarify: There is a bean somewhat that is creating an instance of `LdapUserGraphicalAuthenticationRepository`, right? Inside that bean, you need to build the connection factory (as you're doing above in the ctor) and then pass to the ctor here as an argument.

This is so that,

1. Building "bean" instance should happen during the configuration of the bean itself.
2. Doing this allows you to delete the *manual* ctor and use lombok directly to generate it. Fewer LOC.
3. Makes it easier to write unit tests, in case something has to be mocked down the line. 

I think the same pattern more or less applies to other areas in the patch.